x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/WatcherLifeCycleService.java

```diff
@@ -90,7 +90,7 @@ public void clusterChanged(ClusterChangedEvent event) {
         boolean isWatcherStoppedManually = isWatcherStoppedManually(event.state());
         // if this is not a data node, we need to start it ourselves possibly
         if (event.state().nodes().getLocalNode().isDataNode() == false &&
-            isWatcherStoppedManually == false && this.state.get() == WatcherState.STOPPED) {
+            isWatcherStoppedManually == false && (this.state.get() == WatcherState.STOPPED || this.state.get() == WatcherState.STOPPING)) {
```

Comment:
how about inlining the `||` check into the `WatcherState` class so you can reuse it below?

Reply:
I moved it to a variable inside this method to DRY it up a bit. It felt abit too much to move to `WatcherState`